pkg/executables/executables.go

```diff
@@ -61,47 +65,54 @@ func (e *linuxDockerExecutable) workingDirectory() (string, error) {
 
 func (e *linuxDockerExecutable) Execute(ctx context.Context, args ...string) (bytes.Buffer, error) {
 	var stdout bytes.Buffer
-	if command, err := e.buildCommand(map[string]string{}, e.cli, args...); err != nil {
+	if command, containerName, err := e.buildCommand(map[string]string{}, e.cli, args...); err != nil {
 		return stdout, err
 	} else {
+		defer executeCleanup(ctx, containerName)
 		return execute(ctx, "docker", nil, command...)
 	}
 }
 
 func (e *linuxDockerExecutable) ExecuteWithStdin(ctx context.Context, in []byte, args ...string) (bytes.Buffer, error) {
 	var stdout bytes.Buffer
-	if command, err := e.buildCommand(map[string]string{}, e.cli, args...); err != nil {
+	if command, containerName, err := e.buildCommand(map[string]string{}, e.cli, args...); err != nil {
 		return stdout, err
 	} else {
+		defer executeCleanup(ctx, containerName)
 		return execute(ctx, "docker", in, command...)
 	}
 }
 
-func (e *linuxDockerExecutable) buildCommand(envs map[string]string, cli string, args ...string) ([]string, error) {
+func (e *linuxDockerExecutable) buildCommand(envs map[string]string, cli string, args ...string) ([]string, string, error) {
 	directory, err := e.workingDirectory()
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
 
 	var envVars []string
 	for k, v := range envs {
 		envVars = append(envVars, "-e", fmt.Sprintf("%s=%s", k, v))
 	}
+	containerName := containerNamePrefix + cli + "_" + time.Now().Format("2006-01-0215.04.05.000000")
```

Comment:
What if we use epoch timestamp? I am not sure how readable this is, but epoch would require a converter to actually get the time. This is also worth a read in the case that two cli's were running at the same time and accidentally picks the same time: https://stackoverflow.com/a/54491783